### PR TITLE
Add JUnit tests for TaskManager class, testing basic functionality like adding. updating and deleting tasks

### PR DESCRIPTION
**Pull request by Aak242**

### CODE REVIEW

File: src/test/java/data/TaskManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    void addTask_validInput_addsTask() {
+        // Arrange
+        LocalDate date = LocalDate.now();
+        String taskDescription = "Test task";
+
+        // Act
+        addTask(date, taskDescription);
+
+        // Assert
+        assertTrue(taskManager.getTasksForDate(date).contains(taskDescription));
```
Comment (clarencepohh) [inline src/test/java/data/TaskManagerTest.java:35]: Clear and concise. LGTM

**Review by NicholasTanYY — COMMENTED** [review]
LGTM! Ready for merging.

**Review by clarencepohh — COMMENTED** [review]
All tests look functional and useful. I like how you clearly phrase what's happening in the test. 

Might want to look into why the deleteTask_validInput_deletesTask() is failing in CI

File: src/test/java/data/TaskManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+class TaskManagerTest {
+    private TaskManager taskManager;
+
+    @BeforeEach
+    void setUp() {
+        taskManager = new TaskManager();
+    }
```
Comment (NicholasTanYY) [inline src/test/java/data/TaskManagerTest.java:23]: Add this:

@AfterEach
void resetTaskManager() {
    taskManager = null;
}

hopefully resolves the CI failure

**Comment by clarencepohh** [comment]
Will merge to facilitate testing on local machine

**Pull request merged**